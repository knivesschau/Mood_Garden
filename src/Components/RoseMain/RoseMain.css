/* Mobile styles first, 320px */
.Rose_Expanded {
    margin-top: .25em;
}

.Rose_Image_Entry > img {
    position: relative;
    top: 1.5em;
    left: 1.5em;
    animation: rotation 3.5s infinite linear;
    margin: -2em;
}

@keyframes rotation {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(359deg);
    }
}

@media only screen and (min-device-width: 375px) {
    .Rose_Expanded {
        margin-top: 1em;
        margin-left: 1em;
        margin-right: 1em;
    }

    .Rose_Image_Entry > img {
        top: 1.75em;
    }
}

@media only screen and (min-device-width: 414px) {
    .Rose_Image_Entry > img {
        top: 2.50em;
    }
}

@media only screen and (min-device-width: 768px) {
    .Rose_Image_Entry > img {
        top: 4em;
        left: 4em;
    }
}

@media only screen and (min-device-width: 1000px) {
    .Rose_Image_Entry > img[style] {
        top: 4em;
        left: 8em;
        width: 10% !important;
    }

    .Rose_Main {
        margin-top: 20em;
    }
}

@media only screen and (min-device-width: 1200px) {
    .Rose_Image_Entry > img[style] {
        top: 5em;
        left: 14em;
    }
}

@media only screen and (min-device-width: 1800px) {
    .Rose_Image_Entry > img[style] {
        left: 25em;
        width: 8% !important;
    }
}