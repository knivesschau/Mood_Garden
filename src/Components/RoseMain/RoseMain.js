import React, {Component} from 'react';
import moodGardenContext from '../../moodGardenContext';
import RoseNav from '../RoseNav/RoseNav';
import Rose from '../Rose/Rose';
import './RoseMain.css';

export default class RoseMain extends Component {
    static defaultProps = {
        match: {
            params: {}
        }
    }

    static contextType = moodGardenContext;

    handleDeleteRose = id => {
        this.props.history.push(`/your-garden`)
    }

    render() {
        const {roses=[]} = this.context; 
        const {id} = this.props.match.params;
        const roseEntry = roses.find(rose => rose.id === parseInt(id)) || {};

        console.log(roseEntry);
        
        return (
            <section className="View_Rose">
                <nav role="navigation" className="Rose_Nav">
                    <RoseNav/>
                </nav>
                
                    <Rose 
                        id={roseEntry.id}
                        rose={roseEntry.rose}
                        thorn={roseEntry.thorn}
                        bud={roseEntry.bud}
                        color={roseEntry.color}
                        entry_date={roseEntry.entry_date}
                        onDeleteRose={this.handleDeleteRose}
                    />
            </section>
        )
    }
}