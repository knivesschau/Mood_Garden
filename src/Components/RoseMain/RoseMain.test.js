import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import RoseMain from './RoseMain';

describe ('RoseMain Component', () => {
    it ('renders the complete RoseMain Component and all nested Components', () => {
        const wrapper = shallow(<RoseMain/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})