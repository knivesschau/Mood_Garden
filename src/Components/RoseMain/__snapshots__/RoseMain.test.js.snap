// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoseMain Component renders the complete RoseMain Component and all nested Components 1`] = `
<section
  className="View_Rose"
>
  <nav
    className="Rose_Nav"
    role="navigation"
  >
    <RoseNav />
  </nav>
  <Rose
    onDeleteRose={[Function]}
  />
</section>
`;
